clients/rest-client-generator/src/main/java/uk/gov/justice/services/clients/rest/generator/strategy/ClientGenerationStrategy.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.justice.services.clients.rest.generator.strategy;
+
+import uk.gov.justice.raml.common.mapper.ActionMapping;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.raml.model.ActionType;
+import org.raml.model.MimeType;
+
+public interface ClientGenerationStrategy {
+
+    /**
+     * Returns the list of {@link ActionMapping} for this generation strategy.
+     *
+     * @param description the description from a RAML Action
+     * @return a list of {@link ActionMapping}
+     */
+    List<ActionMapping> listOfActionMappings(final String description);
+
+    /**
+     * Returns the {@link ActionMapping} for this generation strategy.
+     *
+     * @param actionMappings the list of {@link ActionMapping}
+     * @param mimeType       the MimeType to use for the lookup
+     * @param httpMethod     the ActionType to use for the lookup
+     * @return the {@link ActionMapping}
+     */
+    ActionMapping mappingOf(final Collection<ActionMapping> actionMappings, final MimeType mimeType, final ActionType httpMethod);
+
+    /**
+     * Returns the handles annotation value for this generation strategy.
+     *
+     * @param mapping the {@link ActionMapping} to use.
+     * @param header  the header to use
+     * @return the handles annotation value
+     */
+    String handlesValue(final ActionMapping mapping, final String header);
```

Comment:
Do we need header here? - is it used in other implementations? suggests the contract definition is not quite correct...